xbmc/video/VideoDatabase.cpp

```diff
@@ -8474,13 +8474,28 @@ void CVideoDatabase::CleanDatabase(CGUIDialogProgressBarHandle* handle, const st
     std::string strIds;
     while (!m_pDS->eof())
     {
-      auto pathsDeleteDecision = pathsDeleteDecisions.find(m_pDS->fv(0).get_asInt());
       // Check if we have a decision for the parent path
+      auto pathsDeleteDecision = pathsDeleteDecisions.find(m_pDS->fv(0).get_asInt());
       auto pathsDeleteDecisionByParent = pathsDeleteDecisions.find(m_pDS->fv(2).get_asInt());
-      if (((pathsDeleteDecision != pathsDeleteDecisions.end() && pathsDeleteDecision->second) ||
-           (pathsDeleteDecision == pathsDeleteDecisions.end() && !CDirectory::Exists(m_pDS->fv(1).get_asString(), false))) &&
-          ((pathsDeleteDecisionByParent != pathsDeleteDecisions.end() && pathsDeleteDecisionByParent->second) ||
-           (pathsDeleteDecisionByParent == pathsDeleteDecisions.end())))
+
+      bool hasDecision = pathsDeleteDecision != pathsDeleteDecisions.end();
+      bool hasDecisionForParent = pathsDeleteDecisionByParent != pathsDeleteDecisions.end();
+      
+      std::string itemPath = m_pDS->fv( 1 ).get_asString();
+
+      // add the path to the list of paths to be deleted if
+      //   it has been handled before and should be deleted
+      //   OR
+      //   it hasn't been handled before and doesn't exist
+      // AND
+      //   its parent path has been handled before and should be deleted
+      //   OR
+      //   its parent path hasn't been handled before
+      if( 
+          ( ( hasDecision && pathsDeleteDecision->second ) || ( !hasDecision && !CDirectory::Exists( itemPath, false ) ) ) 
+          &&
+          ( ( hasDecisionForParent && pathsDeleteDecisionByParent->second ) || !hasDecisionForParent ) 
+        )
```

Comment:
It's an edgecase and matter of taste, but given our coding guidelines, I'd prefer if you keep the format of the if statement like it was before.
Normally I would suggest to move it into it's own helper function to get rid of this big statements, but given the count of parameters you would need, it won't help much.